### ISSUE TITLE
Nightly OSX release tests fail test_memory_tools

### ISSUE THREAD

**Issue by clalancette** [ros2/build_farmer#137]
The build from [last night](https://ci.ros2.org/view/nightly/job/nightly_osx_ospl_release/47) failed the memory tests on OSX release mode:

```
Running main() from gtest_main.cc
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestMemoryTools
[ RUN      ] TestMemoryTools.test_example
/Users/osrf/jenkins-agent/workspace/nightly_osx_ospl_release/ws/build/test_osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/src/gtest.cc:683: Failure
Expected: 1 non-fatal failure
  Actual: 0 failures
/Users/osrf/jenkins-agent/workspace/nightly_osx_ospl_release/ws/build/test_osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/src/gtest.cc:683: Failure
Expected: 1 non-fatal failure
  Actual: 0 failures
[  FAILED  ] TestMemoryTools.test_example (1 ms)
[----------] 1 test from TestMemoryTools (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestMemoryTools.test_example
```

In https://github.com/osrf/osrf_testing_tools_cpp/pull/8 and https://github.com/osrf/osrf_testing_tools_cpp/pull/9, I actually did some work to improve this, but it is still not enough; that function is still getting optimized away sometimes.  I have a change that should really fix this, which I'll post soon.

**Comment by clalancette** [comment]
PR: https://github.com/osrf/osrf_testing_tools_cpp/pull/11

**Comment by clalancette** [comment]
Fixed by osrf/osrf_testing_tools_cpp#11, closing.
